src/region/RegionListManager.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import BaseManager, { Options, Props } from '../common/BaseManager';
+import RegionAnnotationsContainer from './RegionAnnotationsContainer';
+
+export default class RegionListManager implements BaseManager {
+    location: number;
+
+    reactEl: HTMLElement;
+
+    constructor({ location = 1, referenceEl }: Options) {
+        this.location = location;
+        this.reactEl = this.insert(referenceEl);
+    }
+
+    destroy(): void {
+        ReactDOM.unmountComponentAtNode(this.reactEl);
+
+        this.reactEl.remove();
+    }
+
+    exists(parentEl: HTMLElement): boolean {
+        return parentEl.contains(this.reactEl);
+    }
+
+    insert(referenceEl: HTMLElement): HTMLElement {
+        // Find the nearest applicable reference and document elements
+        const documentEl = referenceEl.ownerDocument || document;
+        const parentEl = referenceEl.parentNode || documentEl;
+
+        // Construct a layer element where we can inject a root React component
+        const rootLayerEl = documentEl.createElement('div');
```

Comment:
We're not seeing as much variation between managers as I expected, which is leading to a lot of duplicate code. We should consider extracting it to a base class or a helper function or something. It can wait for a future change, though.